scripts/dashboard-esbuild.ts

```diff
@@ -0,0 +1,427 @@
+// Node.js-based dashboard build using Group Income's Vue esbuild plugin approach
+// Uses Node.js to avoid Deno compatibility issues with @vue/component-compiler
+
+import { sassPlugin } from 'esbuild-sass-plugin'
+import fs from 'node:fs'
+import { mkdir, readFile, readdir, writeFile } from 'node:fs/promises'
+import { basename, dirname, join, relative, resolve } from 'node:path'
+import process from 'node:process'
+import { fileURLToPath } from 'node:url'
+import componentCompiler, { type SFCCompiler, type StyleCompileResult, type DescriptorCompileResult } from 'npm:@vue/component-compiler'
+import esbuild from 'npm:esbuild'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+// Fix for __proto__ issue with buble (used by @vue/component-compiler)
+// See: https://github.com/denoland/deno/issues/20618
+if (({} as Record<PropertyKey, unknown>).__proto__ !== Object.prototype) {
+  Object.defineProperty(Object.prototype, '__proto__', {
+    get () {
+      return Object.getPrototypeOf(this)
+    },
+    set (v) {
+      Object.setPrototypeOf(this, v)
+    }
+  })
+}
+
+// Alias replacer function (adapted from Group Income's utils.js)
+function createAliasReplacer (aliases: Record<string, string>) {
+  if (Object.keys(aliases).some(alias => alias.includes('/'))) {
+    throw new Error('Path aliases may not include slash characters.')
+  }
+
+  const cwd = process.cwd()
+  const escapeForRegExp = (string: string) => string.replace(/[.*+?^${}()|\\[\]]/g, '\\$&')
+  const escapedAndSortedAliases = Object.keys(aliases).map(escapeForRegExp).sort().reverse()
+  const re = new RegExp(
+    `(?:^import[ (]|\\bimport[ (]|import .+? from |^\\} from )['"](${ escapedAndSortedAliases.join('|')})(?:['"]|/[^'"]+?['"])`,
+    'gm'
+  )
+
+  return function aliasReplacer ({ path, source }: { path: string, source: string }) {
+    const relativeDirPath = relative(dirname(path), cwd)
+
+    return source.replace(re, (match, capture) => {
+      const resolvedPathSegment = aliases[capture]
+      const replacement = join(relativeDirPath, resolvedPathSegment)
+        .replace(/\\/g, '/')
+
+      return match.replace(capture, replacement)
+    })
+  }
+}
+
+// Copy and update index.html with correct asset references
+async function copyAndUpdateIndexHtml (outDir: string) {
+  try {
+    const sourceHtml = join(__dirname, '../src/serve/dashboard/index.html')
+    const targetHtml = join(outDir, 'index.html')
+
+    // Read the source HTML
+    let htmlContent = await readFile(sourceHtml, 'utf8')
+
+    // Update asset references
+    htmlContent = htmlContent.replace(
+      '<script type="module" src="/main.js"></script>',
+      '<script type="module" src="/assets/js/main.js"></script>\n  <link rel="stylesheet" href="/assets/css/main.css">'
+    )
```

Comment:
What is this and why is it here?

It doesn't make sense IMO for a build script for the dashboard to be making this type of change here.

Reply:
Yes, that's my mistake.
This is not needed, `index.html` in dashboard already have these imports.